File: lib/aws/ec2/internetGateway.ts
// Copyright 2016 Pulumi, Inc. All rights reserved.

import * as cloudformation from '../cloudformation';

// An Internet gateway enables your instances to connect to the Internet through the Amazon EC2 edge network.
// @name: aws/ec2/internetGateway
// @website: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-internet-gateway.html 
export class InternetGateway extends cloudformation.Resource {
    constructor(name: string, args: InternetGatewayArgs) {
        super({
            name: name,
            resource: "AWS::EC2::InternetGateway",
            properties: args,
        });
    }
}

export interface InternetGatewayArgs extends cloudformation.TagArgs {
}

File: lib/aws/idl/ec2/internetGateway.go
// Copyright 2016-2017, Pulumi Corporation.  All rights reserved.

package ec2

import (
	"github.com/pulumi/lumi/pkg/resource/idl"
)

// An Internet gateway enables your instances to connect to the Internet through the Amazon EC2 edge network.  See
// http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-internet-gateway.html.
type InternetGateway struct {
	idl.NamedResource
}
